src/main/kotlin/no/nav/syfo/saf/SafJournalpostClient.kt

```diff
@@ -0,0 +1,44 @@
+package no.nav.syfo.saf
+
+import io.ktor.client.HttpClient
+import io.ktor.client.call.receive
+import io.ktor.client.request.header
+import io.ktor.client.request.post
+import io.ktor.client.statement.HttpStatement
+import io.ktor.http.HttpHeaders
+import io.ktor.http.HttpStatusCode
+import no.nav.syfo.graphql.model.GraphQLResponse
+import no.nav.syfo.log
+import no.nav.syfo.saf.model.GetJournalpostRequest
+import no.nav.syfo.saf.model.GetJournalpostVariables
+import no.nav.syfo.saf.model.JournalpostResponse
+
+class SafJournalpostClient(
+    private val httpClient: HttpClient,
+    private val basePath: String,
+    private val graphQlQuery: String
+) {
+
+    suspend fun getJournalpostMetadata(journalpostId: String, token: String): GraphQLResponse<JournalpostResponse>? {
+
+        log.info("Henter journalpostmetadata for $journalpostId")
+
+        val getJournalpostRequest = GetJournalpostRequest(query = graphQlQuery, variables = GetJournalpostVariables(journalpostId))
+        val httpResponse = httpClient.post<HttpStatement>(basePath) {
+            body = getJournalpostRequest
+            header(HttpHeaders.Authorization, "Bearer $token")
```

Comment:
Funker det å bruke innlogget brukers token, selv om det er skopet til smregistrering-backend..?

Reply:
Ja, men jeg kan altlids lage et annet for å få det mer "riktig" :)

Reply:
Ja det ser sånn ut! Jeg er litt usikker på hva vi ev. skal scope det til hvis vi skal endre det?

Reply:
Da må det evt scopes til SAF sin clientid. Men, hvis det virker sånn det er nå er det sikkert fint sånn :)